src/ApplicationInsights.Kubernetes/ContainerIdProviders/MountInfoContainerIdProvider.cs

```diff
@@ -0,0 +1,65 @@
+#nullable enable
+
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.ApplicationInsights.Kubernetes.Debugging;
+
+namespace Microsoft.ApplicationInsights.Kubernetes.ContainerIdProviders;
+
+internal class MountInfoContainerIdProvider : IContainerIdProvider
+{
+    private const string InfoFilePath = "/proc/self/mountinfo";
+    private const string MatchPattern = @"\/docker\/containers\/(.*?)\/";
```

Comment:
Is this going to work for containerD and other container runtimes? Docker is not present by default with latest version of Kubernetes...

In general, how reliable is discovering container ID based on mount info? Is there a K8s spec/design doc that can be referenced here?

Reply:
@karolz-ms, I didn't know there's `containerD` and other runtimes (well, now I think about it, why isn't there? :-)). That is very good point. I need some time to learn it. Tracking by #275 .

I assume you are okay to unblock this change since:

1. It works with docker engine;
2. `ContainerIdProvider` is designed in a way of extensible - mountinfo provider is just one node on the change.

Reply:
Does it work with AKS? ☺️

Reply:
We synced offline. I am going to add another iteration to handle containerD as the runtime and we will leave other engines on request bases unless the usage changes.